deployment/stacks/api.ts

```diff
@@ -442,10 +442,12 @@ export function createApiComponents(props: CreateApiComponentsProps) {
     },
     "REGIONAL"
   );
-  new wafv2.CfnWebACLAssociation(scope, "WebACLAssociation", {
-    resourceArn: api.deploymentStage.stageArn,
-    webAclArn: waf.webAcl.attrArn,
-  });
+  if (waf.webAcl) {
```

Comment:
Is that true only for non-localstack? If so I wonder if it'd be more clear to do the `isLocalStack` check vs this one. Or even just a little comment explaining would suffice.

Reply:
Yes, with localstack because we don't have pro. We can either if every place we call the WAF construct or I thought it would be more efficient to just make the WAF construct do nothing when its localstack... well... then the WAF construct's properties need to have an undefined possibility... that's why this if (waf.webAcl) needs to be here to make sure its not undefined. If we don't have to add the undefined potential, we don't need to add this if. It may be better to huddle to discuss.